`TestDiscoveryDatabase/discover_AWS_database_with_assumed_role` flakiness

## Failure

First detected August 17, 2023 (in case that helps identify the cause).

#### Link(s) to logs

- https://github.com/gravitational/teleport.e/actions/runs/6157877604/job/16709543228#step:7:394

#### Relevant snippet


```
=== CONT  TestDiscoveryDatabase/discover_AWS_database_with_assumed_role
    discovery_test.go:1494: 
        	Error Trace:	/__w/teleport.e/teleport.e/lib/srv/discovery/discovery_test.go:1494
        	Error:      	Condition never satisfied
        	Test:       	TestDiscoveryDatabase/discover_AWS_database_with_assumed_role
time="2023-09-12T11:53:34Z" level=info msg="Watcher done." kind=db trace.component="discovery:service"
--- FAIL: TestDiscoveryDatabase/discover_AWS_database_with_assumed_role (3.00s)
time="2023-09-12T11:53:34Z" level=warning msg="Mux serve failed." error="listener is closed\n\taccept tcp 127.0.0.1:42719: use of closed network connection" trace.component=auth
```

--- a/lib/srv/discovery/discovery_test.go
+++ b/lib/srv/discovery/discovery_test.go
@@ -1493,11 +1493,11 @@ func TestDiscoveryDatabase(t *testing.T) {
 			if tc.wantEvents > 0 {
 				require.Eventually(t, func() bool {
 					return reporter.EventsCount() == tc.wantEvents
-				}, 100*time.Millisecond, 10*time.Millisecond)
+				}, time.Second, 100*time.Millisecond)
 			} else {
 				require.Never(t, func() bool {
 					return reporter.EventsCount() != 0
-				}, 100*time.Millisecond, 10*time.Millisecond)
+				}, time.Second, 100*time.Millisecond)
 			}
 		})
 	}

